code/src/java/pcgen/core/PlayerCharacter.java

```diff
@@ -7765,7 +7765,7 @@ public boolean doLevelAbilities()
 
 	public void adjustAbilities(final Category<Ability> aCategory, final BigDecimal arg)
 	{
-		if (arg.equals(BigDecimal.ZERO))
+		if (arg.compareTo(BigDecimal.ZERO) == 0)
```

Comment:
The method BigDecimal.equals() takes scale into consideration. For example BigDecimal("0.00").equals(BigDecimal.ZERO) == false. I don't know all the callsites here, but seeing .equals() on a BigDecimal in code smell